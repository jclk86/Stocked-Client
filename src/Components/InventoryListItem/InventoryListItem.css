.item_card {
  background-color: rgb(241, 235, 205);
  width: 250px;
  flex-basis: 20%;
  padding: 10px;
  position: relative;
  margin: 5px;
  text-align: center;
  text-decoration: none;
  overflow-wrap: break-word;
  border-radius: 1px;
  box-shadow: 1px 1px 1px 1px rgb(61, 61, 61);
}

.inventory_image {
  max-width: 100%;
  max-height: 100%;
}

.card_title {
  font-size: 22px;
  line-height: 26px;
  margin-top: 0;
  margin-bottom: 4px;
  text-decoration: underline;
}

.card_desc {
  margin-top: 3px;
  font-size: 16px;
  margin-bottom: 3px;
  color: rgb(80, 78, 78);
  font-weight: bold;
}

.restock_message {
  margin-top: 0;
  color: red;
  font-weight: bold;
}

.card_label_qty {
  color: black;
  font-weight: bold;
}

.qty_no_style {
  color: rgb(80, 78, 78);
  margin-top: 0;
  font-weight: bold;
}

.card_title_total_cost {
  margin-top: 5px;
  margin-bottom: 0px;
}

.card_cost {
  margin-top: 0;
  margin-bottom: 0;
  padding-bottom: 0;
  font-weight: bold;
  color: rgb(80, 78, 78);
}

/* Mobile */
@media only screen and (max-width: 600px) {
  .mobile_hidden {
    display: none;
  }

  .item_card {
    flex-basis: 45%;
  }
}
