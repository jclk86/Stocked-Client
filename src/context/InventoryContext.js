import React, { Component } from "react";
import { tags, inventory, units, user } from "../data";

const InventoryContext = React.createContext({
  inventoryList: [],
  tagsList: [],
  unitsList: [],
  error: null,
  addInventoryItem: () => {},
  setInventoryList: () => {},
  updateInventoryItem: () => {},
  deleteInventoryItem: () => {},
  clearInventoryItem: () => {},
  setError: () => {},
  clearError: () => {}
});
export default InventoryContext;

export class InventoryProvider extends Component {
  state = {
    inventoryList: inventory, // change to nullInventory,
    error: null
  };

  setError = error => {
    console.error(error);
    this.setState({ error }); // these must be set with conditionals. on the component rendering area you want to render error message if not rendered. Context becomes errorboundary, Not need fo errorBoundary component
  };

  clearError = () => {
    this.setState({ error: null });
  };

  setInventoryList = inventoryList => {
    this.setState({ inventoryList });
  };

  addInventoryItem = item => {
    this.setState({ inventoryList: [...this.state.inventoryList, item] });
  };

  updateInventoryItem = updatedItem => {
    this.setState({
      inventoryList: this.state.inventoryList.map(item =>
        item.itemId !== updatedItem.itemId ? item : updatedItem
      )
    });
  };

  deleteInventoryItem = deletedItemId => {
    this.setState({
      inventoryList: this.state.inventoryList.filter(
        item => item.itemId !== parseInt(deletedItemId)
      )
    });
  };

  // clearArticle = () => {
  //   this.setInventoryList(nullInventory)
  //   this.setComments([])
  // }

  render() {
    const contextValue = {
      inventoryList: this.state.inventoryList,
      date: new Date(),
      userId: 1,
      tagsList: tags,
      unitsList: units,
      addInventoryItem: this.addInventoryItem,
      setInventorylist: this.setInventoryList,
      updateInventoryItem: this.updateInventoryItem,
      deleteInventoryItem: this.deleteInventoryItem
    };
    return (
      <InventoryContext.Provider value={contextValue}>
        {this.props.children}
      </InventoryContext.Provider>
    );
  }
}
