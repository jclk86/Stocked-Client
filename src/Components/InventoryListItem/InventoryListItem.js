import React, { Component } from "react";
import "./InventoryListItem.css";
import { NavLink, withRouter } from "react-router-dom";
import PropTypes from "prop-types";

class InventoryListItem extends Component {
  renderRestockMessage = (itemQuantity, itemUnit) => {
    return parseInt(itemQuantity) === 0 ? (
      <span className="restock_message">None left. Restock.</span>
    ) : (
      <span className="">
        {itemQuantity} {itemUnit}
      </span>
    );
  };

  renderTotalCost = (itemQuantity, itemCost) => {
    const calc = itemQuantity * itemCost;
    const totalCost = calc.toFixed(2);
    return `$  ${totalCost}`;
  };

  render() {
    const { item } = this.props;
    return (
      <NavLink
        role="navigation"
        to={`/${item.user_id}/inventory/edit-item/${item.item_id}`}
        className="item_card"
      >
        <div role="img" aria-label={`A ${item.name}`}>
          <img
            src={item.image_url}
            className="inventory_image"
            alt={`${item.name}`}
          ></img>
        </div>

        <h2 className="card_title">{item.name}</h2>
        <p className="card_desc"> {item.desc}</p>
        <div
          className={item.quantity === 0 ? "restock_message" : "qty_no_style"}
        >
          {" "}
          <label className="card_label_qty">Qty:</label>{" "}
          {this.renderRestockMessage(item.quantity, item.unit)}
        </div>
        <h4 className="card_title_total_cost">Total Cost: </h4>
        <p className="card_cost">
          {this.renderTotalCost(item.quantity, item.cost_per_unit)}
        </p>
      </NavLink>
    );
  }
}

InventoryListItem.defaultProps = {
  image:
    "https://images.pexels.com/photos/1907642/pexels-photo-1907642.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
};

InventoryListItem.propTypes = {
  item: PropTypes.shape({
    user_id: PropTypes.number,
    item_id: PropTypes.number,
    name: PropTypes.string,
    image: PropTypes.string,
    desc: PropTypes.string,
    quantity: PropTypes.number,
    cost_per_unit: PropTypes.string
  })
};

export default withRouter(InventoryListItem);
