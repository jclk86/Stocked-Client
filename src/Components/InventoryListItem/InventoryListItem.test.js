import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import InventoryListItem from "./InventoryListItem";
import InventoryContext from "../../context/InventoryContext";

describe(`InventoryListItem Component`, () => {
  const contextType = InventoryContext;
  const item = {
    itemId: 1
  };
  it("renders inventory list without issue", () => {
    const wrapper = shallow(
      <InventoryListItem value={contextType} item={item}></InventoryListItem>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
