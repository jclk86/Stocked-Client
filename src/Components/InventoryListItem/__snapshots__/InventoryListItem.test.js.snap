// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryListItem Component renders inventory list without issue 1`] = `
<NavLink
  className="item_card"
  role="navigation"
  to="/edit-item/1"
>
  <div
    aria-label="A undefined"
    role="img"
  >
    <img
      alt="undefined"
      className="inventory_image"
    />
  </div>
  <h2 />
  <p>
     
  </p>
  <p
    className=""
  >
     
    Qty: 
  </p>
  <p>
    Total Cost: 
    $ NaN
  </p>
</NavLink>
`;
